### PR TITLE
JBPM-5462: Fixing some memory leaks

### PR DESCRIPTION
**Pull request by romartin**
Hey @hasys @manstis 

Here are some bug fixings for some of the memory leaks on the lienzo side. 

Related PRs:
- https://github.com/kiegroup/lienzo-tests/pull/18
- https://github.com/kiegroup/kie-wb-common/pull/1685

Thanks!

### CODE REVIEW

File: src/main/java/com/ait/lienzo/client/core/image/ImageLoader.java [hunk 1]
```diff
@@ -71,7 +86,53 @@ public final void onError(final ErrorEvent event)
...
+    public ImageLoader(final ImageResource resource,
+                       final Image image)
+    {
+        final ImageElement element = ImageElement.as(image.getElement());
+
+        image.setVisible(false);
+
+        final HandlerRegistrationManager m_HandlerRegManager = new HandlerRegistrationManager();
+
+        m_HandlerRegManager.register(
```
Comment (manstis) [inline src/main/java/com/ait/lienzo/client/core/image/ImageLoader.java]: Some crazy formatting here.. 

I know ```ahome-it``` doesn't follow ```KIE``` but formatting is probably not consistent with ```ahome-it``` either.

File: src/main/java/com/ait/lienzo/client/widget/LienzoHandlerManager.java [hunk 2]
```diff
@@ -140,9 +143,27 @@ public LienzoHandlerManager(final LienzoPanel lienzo)
...
+    {
+        if (null != m_mediators)
+        {
+            while (null != m_mediators.pop()) {}
+        }
+        if (null != m_touches)
+        {
+            m_touches.clear();
+        }
+        handlerRegistrationManager.removeHandler();
```
Comment (manstis) [inline src/main/java/com/ait/lienzo/client/widget/LienzoHandlerManager.java]: Nit picking... ```handlerRegistrationManager.destroy()``` would have possibly been clearer/more consistent with this method name (and would have avoided me having to lookup what ```handerRegistrationManager.removeHandler()``` did - which is just to call ```destroy()``` ;-)

File: src/main/java/com/ait/lienzo/client/core/image/ImageProxy.java [hunk 3]
```diff
@@ -34,13 +34,17 @@
...
+import com.google.gwt.user.client.ui.Image;
+import com.google.gwt.user.client.ui.RootPanel;
 
 /**
  * ImageProxy is used by {@link AbstractImageShape} to load and draw the image.
  */
 public class ImageProxy<T extends AbstractImageShape<T>> implements ImageDataFilterable<ImageProxy<T>>
 {
-    private T                          m_image;
+    private final T                          m_image;
```
Comment (jomarko) [inline src/main/java/com/ait/lienzo/client/core/image/ImageProxy.java]: Could we indent `m_image` to same distance as other class attributes?

File: src/main/java/com/ait/lienzo/client/core/image/ImageProxy.java [hunk 4]
```diff
@@ -670,6 +680,22 @@ public BoundingBox getBoundingBox()
...
+        m_image.removeFromParent();
+        m_jsimg.removeFromParent();
+        m_normalImage.clear();
+        m_filterImage.clear();
+        m_selectImage.clear();
+        m_filters.clearFilters();
+        m_handler = null;
+        m_obounds = null;
+        m_img = null;
+        m_jsimg = null;
```
Comment (jomarko) [inline src/main/java/com/ait/lienzo/client/core/image/ImageProxy.java]: Shouldn't we destroy also `m_ignores` somehow?

File: src/main/java/com/ait/lienzo/client/core/shape/wires/SelectionManager.java [hunk 5]
```diff
@@ -844,6 +843,17 @@ public void clear()
...
 
+        public void destroy()
+        {
+            selectionGroup  = false;
+            m_changed.clear();
+            m_shapes.clear();
+            m_connectors.clear();
+            m_externallyConnected.clear();
+            m_bbox = null;
+            m_selManager.m_startBoundingBox = null;
```
Comment (jomarko) [inline src/main/java/com/ait/lienzo/client/core/shape/wires/SelectionManager.java]: Similar question, looking on list of class memebers, there seems to me more complex objects that should be destroyed somehow? But sure, I know basically nothing about lienzo, so takes this just as question.

File: src/main/java/com/ait/lienzo/client/core/shape/wires/WiresManager.java [hunk 6]
```diff
@@ -261,6 +278,38 @@ public void resetContext() {
         }
     }
 
+    private void destroy() {
```
Comment (hasys) [inline src/main/java/com/ait/lienzo/client/core/shape/wires/WiresManager.java]: lienzo code style missing for this method.

**Comment by hasys** [comment]
It looks like https://github.com/kiegroup/lienzo-tests/pull/18 contains just one small change. Does it mean that all tests are passing with this changes or new changes are not covered?

**Pull request merged**